pkg/user/postgres.go

```diff
@@ -19,13 +20,39 @@ func NewPostgresRepo(db *gorm.DB) Repository {
 // GetUserByEmail returns a user by email
 func (r *repo) GetUserByEmail(email string) (*models.User, error) {
 	// TODO: Implement GetUserByEmail
+	var user models.User
+	result := r.DB.Where("email = ?", email).First(&user)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	if result.RowsAffected == 0 {
+		return nil, nil
```

Comment:
Catching 404 is available via ```gorm.ErrRecordNotFound```

You should return nil, err in this case.

Reply:
I will make the changes